# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/angle/checkout/src/common/PackedEnums.h
# Commit: 13282d7a47a5
# Full Hash: 13282d7a47a57a73be8d060efd9645b4d642f03e
# Author: Miko Mynttinen <mikokm@gmail.com>
# Date: 2019-09-08 09:46:52
# Regressor Bug: 1578576
# File Overlap Count: 4
# Description:
#   Bug 1578576 - Part 2: Update to ANGLE 3865 r=jgilbert
#   
#   Depends on D44561
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D44579
# ==============================================================================

diff -r 3064469c073d -r 13282d7a47a5 gfx/angle/checkout/src/common/PackedEnums.h
--- a/gfx/angle/checkout/src/common/PackedEnums.h	Sat Sep 07 13:36:46 2019 +0000
+++ b/gfx/angle/checkout/src/common/PackedEnums.h	Sat Sep 07 13:38:36 2019 +0000
@@ -269,6 +269,8 @@
 static_assert(ToGLenum(PrimitiveMode::TriangleStripAdjacency) == GL_TRIANGLE_STRIP_ADJACENCY,
               "PrimitiveMode violation");
 
+std::ostream &operator<<(std::ostream &os, PrimitiveMode value);
+
 enum class DrawElementsType : size_t
 {
     UnsignedByte  = 0,
@@ -311,6 +313,8 @@
 ANGLE_VALIDATE_PACKED_ENUM(DrawElementsType, UnsignedShort, GL_UNSIGNED_SHORT);
 ANGLE_VALIDATE_PACKED_ENUM(DrawElementsType, UnsignedInt, GL_UNSIGNED_INT);
 
+std::ostream &operator<<(std::ostream &os, DrawElementsType value);
+
 enum class VertexAttribType
 {
     Byte               = 0,   // GLenum == 0x1400
@@ -367,6 +371,8 @@
 ANGLE_VALIDATE_PACKED_ENUM(VertexAttribType, Fixed, GL_FIXED);
 ANGLE_VALIDATE_PACKED_ENUM(VertexAttribType, Int2101010, GL_INT_2_10_10_10_REV);
 ANGLE_VALIDATE_PACKED_ENUM(VertexAttribType, UnsignedInt2101010, GL_UNSIGNED_INT_2_10_10_10_REV);
+
+std::ostream &operator<<(std::ostream &os, VertexAttribType value);
 }  // namespace gl
 
 namespace egl